apis/Google.Cloud.Logging.Log4Net/Google.Cloud.Logging.Log4Net/GoogleStackdriverAppender.cs

```diff
@@ -1,502 +1,508 @@
+            {
+                labels.Add(nameof(MetaDataType.Identity), loggingEvent.Identity ?? unknown);
+            }
+            if (_withMetaData.Contains(MetaDataType.ThreadName))
+            {
+                labels.Add(nameof(MetaDataType.ThreadName), loggingEvent.ThreadName ?? unknown);
+            }
+            if (_withMetaData.Contains(MetaDataType.UserName))
+            {
+                labels.Add(nameof(MetaDataType.UserName), loggingEvent.UserName ?? unknown);
+            }
+            if (_withMetaData.Contains(MetaDataType.Domain))
+            {
+                labels.Add(nameof(MetaDataType.Domain), loggingEvent.Domain ?? unknown);
+            }
+            if (_withMetaData.Contains(MetaDataType.LoggerName))
+            {
+                labels.Add(nameof(MetaDataType.LoggerName), loggingEvent.LoggerName ?? unknown);
+            }
+            if (_withMetaData.Contains(MetaDataType.Level))
+            {
+                labels.Add(nameof(MetaDataType.Level), loggingEvent.Level?.Name ?? unknown);
+            }
+            TryAddGitRevisionId(labels);
+            foreach (var customLabel in _customLabels)
+            {
+                labels.Add(customLabel.Key, customLabel.Value);
+            }
+            var logEntry = new LogEntry
+            {
+                TextPayload = RenderLoggingEvent(loggingEvent),
+                Severity = s_levelMap[loggingEvent.Level],
+                Timestamp = loggingEvent.TimeStamp.ToTimestamp(),
+                LogName = _logName,
+                Resource = _resource,
+                Labels = { labels },
+            };
+            if (sourceLocation != null)
+            {
+                logEntry.SourceLocation = sourceLocation;
+            }
+            return logEntry;
+        }
+
+        private void TryAddGitRevisionId(Dictionary<string, string> labels)
+        {
+            try
+            {
+                var gitId = SourceContext.AppSourceContext?.Git?.RevisionId;
+                if (!String.IsNullOrWhiteSpace(gitId))
+                {
+                    labels.Add(SourceContext.GitRevisionIdLogLabel, gitId);
+                }
+            }
+            catch (Exception ex) when (
+                ex is SecurityException 
+                || ex is InvalidProtocolBufferException
+                || ex is InvalidJsonException 
+                || ex is UnauthorizedAccessException)
+            {
+                // This is best-effort only, exceptions from reading/parsing the source_context.json are ignored.
+            }
+        }
+
+        private LogEntrySourceLocation BuildLogEntryLocation(LoggingEvent loggingEvent)
+        {
+            string file = null;
+            long? line = null;
+            string function = null;
+            if (loggingEvent.LocationInformation?.FileName != null)
+            {
+                file = loggingEvent.LocationInformation?.FileName;

```

Comment:
It appears the latest version of log4net now has a filename of "?", rather than empty/null, when it doesn't know the filename.
Please insert this code:
``` C#
if (file == "?")
{
    file = null;
}
```
following this assignment of `file`